internal/ingress/controller/template/configmap.go

```diff
@@ -226,6 +229,33 @@ func ReadConfig(src map[string]string) config.Configuration {
 		to.GlobalExternalAuth.AuthSnippet = val
 	}
 
+	if val, ok := conf[globalAuthCacheKey]; ok {
+		delete(conf, globalAuthCacheKey)
+
+		to.GlobalExternalAuth.AuthCacheKey = val
+	}
+
+	// Verify that the configured global external authorization cache duration is valid
+	if val, ok := conf[globalAuthCacheDuration]; ok {
+		delete(conf, globalAuthCacheDuration)
+
+		if len(val) != 0 {
+			durarr := strings.Split(val, ",")
+			for _, duration := range durarr {
+				duration = strings.TrimSpace(duration)
+				if len(duration) > 0 {
+					if !authreq.ValidCacheDuration(duration) {
+						klog.Warningf("Global auth location denied - %v: %s", "invalid cache duration list", duration)
+					} else {
+						cacheDurations = append(cacheDurations, duration)
+					}
+				}
+			}
+		}
```

Comment:
Is there any way to not duplicate this logic? @aledbf do we have means to share code between here and annotation? like a utils package or something

Reply:
I centralized the "parsing" logic for now in https://github.com/kubernetes/ingress-nginx/pull/4278/commits/d6b901419501b9bf711d650ac7dd0d9bee05b180. Let me know if you want it in a utils package. I digged into this topic - it seems there's a bunch of code that is copy/pasted with slight adaptions (e.g. returning vs. logging an error).